hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -412,6 +412,34 @@ private void validateTableIsIcebergTableOrView(
     }
   }
 
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    TableIdentifier baseTableIdentifier = identifier;
+    if (!isValidIdentifier(identifier)) {
+      if (isValidMetadataIdentifier(identifier)) {
+        baseTableIdentifier = TableIdentifier.of(identifier.namespace().levels());
+      } else {
+        return false;
+      }
+    }
+
+    String database = baseTableIdentifier.namespace().level(0);
+    String tableName = baseTableIdentifier.name();
+    try {
+      Table table = clients.run(client -> client.getTable(database, tableName));
+      HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, identifier));
```

Comment:
Here we check if the base table is an Iceberg table, but in case of metadata tables in the error message we print the metadata table name instead of the base table name. Doesn't seem to matter much but I think we should print the base table name here as the check is also performed against the base table.

Reply:
Sure, updated to `baseTableIdentifier` when logging as part of exception message